addons/binding/org.openhab.binding.camera/src/main/java/org/openhab/binding/camera/handler/CameraHandler.java

```diff
@@ -0,0 +1,177 @@
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link CameraHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Thomas Hartwig - Initial contribution
+ */
+public class CameraHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(CameraHandler.class);
+    private final AtomicBoolean refreshInProgress = new AtomicBoolean(false);
+    private final AtomicBoolean initialized = new AtomicBoolean(false);
+    private final ExecutorService serviceCached = Executors.newCachedThreadPool();
+    private String urlSnapshot;
+    private String urlUsername;
+    private String urlPassword;
+    private long polltime_ms = 5000;
+
+    public CameraHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (CHANNEL_IMAGE.equals(channelUID.getId())) {
+            if (command instanceof RefreshType) {
+                refreshData();
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        logger.debug("Initialize thing: {}::{}", getThing().getLabel(), getThing().getUID());
+        urlSnapshot = (String) getConfig().get("urlSnapshot");
+        urlUsername = (String) getConfig().get("urlUsername");
+        urlPassword = (String) getConfig().get("urlPassword");
+        try {
+            Object param = getConfig().get("poll");
+            polltime_ms = (long) (Double.parseDouble(String.valueOf(param)) * 1000);
+        } catch (Exception e1) {
+            logger.warn("could not read poll time from configuration", e1);
+        }
+        logger.debug("Schedule update at fixed rate {} ms.", polltime_ms);
+        if (initialized.compareAndSet(false, true)) {
+            WeakReference<CameraHandler> weakReference = new WeakReference<>(this);
+            serviceCached.submit(new Callable<Void>() {
+                @Override
+                public Void call() throws Exception {
+                    while (weakReference.get() != null) {
+                        refreshData();
+                        Thread.sleep(Math.max(10, polltime_ms));
+                    }
+                    return null;
+                }
+            });
+        }
+        updateStatus(ThingStatus.OFFLINE);
+    }
+
+    private void refreshData() {
+        if (refreshInProgress.compareAndSet(false, true)) {
```

Comment:
If you want to guard against this, good idea, please take a look at `ExpiringCache` its part of Eclipse SmartHome since a few months and it should resolve this way better.